### PR TITLE
use JSXExpressionContainer or JSXText as child (babel only)

### PR DESCRIPTION
**Pull request by texttechne**
Solves #16

### CODE REVIEW

**Comment by AlexGilleran** [comment]
Sorry about the Travis thing, in the middle of setting that up at the moment.

**Comment by texttechne** [comment]
great :+1:

File: babel.js [hunk 1]
```diff
@@ -37,33 +19,32 @@ module.exports = function (babel) {
...
     ifBlock = _.takeWhile(children, notElseTag);
 
     if (ifBlock.length > 1) {
       throwError(errors.MULTIPLE_CHILDREN, node, file);
     } else if (ifBlock.length === 0) {
       throwError(errors.NO_CHILDREN, node, file);
+    } else {
+      ifBlock = ifBlock[0];
     }
+    //ifBlock = ifBlock.length > 1 ?  t.ArrayExpression(ifBlock) : ifBlock[0];
```
Comment (AlexGilleran) [inline babel.js]: Remove this if it's not needed.

File: babel.js [hunk 2]
```diff
@@ -37,33 +19,32 @@ module.exports = function (babel) {
...
 
-    var children = removeLiterals(node.children);
-
-    var notElseTag = function (child) {
-      return child.type !== 'JSXElement' || child.openingElement.name.name !== 'Else';
-    };
-
-    var ifBlock, elseBlock;
+    // normalize JSXText and JSXExpressionContainer to expressions
+    var children = t.react.buildChildren(node);
```
Comment (AlexGilleran) [inline babel.js]: So this basically does what removeLiterals did but built-in to babel? Nice find.

**Comment by AlexGilleran** [comment]
Awesome work, just a couple of things that need cleaning up :).

File: babel.js (hunk 1 above)
Comment (texttechne) [inline babel.js]: ahh, it was a reminder. It's possible to allow for more than one child within control statements. Consequence: We would need an array which leads to extra `<span>`s in some circumstances and React wants `key`s for every item, which could be automatically generated...

File: babel.js (hunk 2 above)
Comment (texttechne) [inline babel.js]: good catch.

File: babel.js (hunk 1 above)
Comment (AlexGilleran) [inline babel.js]: No worries - could you add a comment to that effect above it then?

**Comment by texttechne** [comment]
PR updated

File: babel.js (hunk 1 above)
Comment (texttechne) [inline babel.js]: created an issue for that

**Comment by AlexGilleran** [comment]
Merged, thanks :). Given all the good work you've been doing I've added you as a contributor so you have push access from now on - just make sure you assign an issue to yourself before you start working on it so we don't end up working on the same things simultaneously :).

**Comment by texttechne** [comment]
Thanks :)

**Pull request merged**